actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -163,7 +322,13 @@ def respond_to_missing?(*args)
           end
 
           def method_missing(called, *args, **options, &block)
-            tag_string(called, *args, **options, &block)
+            name = called.to_s.dasherize
+
+            tag_string(name, *args, **options, &block).tap do
+              ActiveSupport::CodeGenerator.batch(TagBuilder, __FILE__, __LINE__) do |code_generator|
+                TagBuilder.define_element(name, code_generator: code_generator, method_name: name.underscore)
+              end
+            end
```

Comment:
I'm happy to remove it. This was an idea that I'd pitched to @rafaelfranca. I'm mostly thinking of custom elements like `<turbo-frame>` or newer elements like `<search>`. Fairly uncommon.

Would it make sense to consider some kind of public interface for consumer applications and gems to define elements, or is that too much work for minor performance improvements?

Reply:
Yeah, I guess I just got bit pretty hard by that pattern in the past.

Would be interesting to bench of the perf difference between method missing and a class_evel really justifies it for the rare elements that aren't defined.

Reply:
For what it's worth, prior implementations never did anything to avoid `method_missing` calls, so doing nothing about it in this diff would maintain the status quo.

Reply:
So maybe we should just keep the method missing without the eval fallback? If turbo wants to register a turbo-frame tag we should allow them to do it with a public API.

Reply:
@rafaelfranca that makes sense to me. Could providing a public interface for configuring the builder be explored in a separate PR?